src/extension.ts

```diff
@@ -713,6 +715,46 @@ class MacroInlayHintsProvider implements vscode.InlayHintsProvider {
     }
 }
 
+class PatternInlayHintsProvider implements vscode.InlayHintsProvider {
```

Comment:
Can we combine this and the macro inlay provider? Seems like they do the same thing, it'd be easier to just have an if statement in the loop to vary which inlay is added.